.clang-format

```diff
@@ -0,0 +1,25 @@
+BasedOnStyle: GNU
+AccessModifierOffset: -4
+AlignAfterOpenBracket: DontAlign
+AlignEscapedNewlines: DontAlign
+AlignOperands: DontAlign
+AllowAllArgumentsOnNextLine: false
+AllowAllParametersOfDeclarationOnNextLine: false
+AllowShortIfStatementsOnASingleLine: false
+BreakBeforeBinaryOperators: None
+BreakBeforeBraces: Stroustrup
+IndentWidth: 4
+SortIncludes: Never
+SpaceAfterCStyleCast: true
+SpaceBeforeParens: Never
+SpacesInParentheses: true
+TabWidth: 4
+UseTab: Always
+# Prevent clang-format from adding a space after the return statement, i.e:
+# `return( TRUE );` -> `return ( TRUE );`
+WhitespaceSensitiveMacros: [ 'return' ]
+# Our column limit is actually 80, but setting that results in clang-format
+# making a lot of dubious hanging-indent choices; disable it and assume the
+# developer will line wrap appropriately. clang-format will still check
+# existing hanging indents.
+ColumnLimit: 0
```

Comment:
This idea is borrowed from GLib:
https://github.com/GNOME/glib/blob/9f108815740c7bb64bd8fae80d89161ffe49add6/.clang-format#L7-L11